spec/std/io/buffered_spec.cr

```diff
@@ -284,19 +284,86 @@ describe "IO::Buffered" do
     str.to_s.should eq("abcd")
   end
 
-  it "syncs" do
-    str = IO::Memory.new
+  describe "sync" do
+    it "syncs (write)" do
+      str = IO::Memory.new
 
-    io = BufferedWrapper.new(str)
-    io.sync?.should be_false
+      io = BufferedWrapper.new(str)
+      io.sync?.should be_false
+
+      io.sync = true
+      io.sync?.should be_true
+
+      io.write_byte 1_u8
+
+      str.rewind
+      str.read_byte.should eq(1_u8)
+    end
+
+    it "works with IO#read" do
+      str = IO::Memory.new "abc"
+
+      io = BufferedWrapper.new(str)
+      io.sync?.should be_false
+
+      io.sync = true
+      io.sync?.should be_true
+
+      byte = Bytes.new(1)
+      io.read_fully(byte)
+      byte[0].should eq('a'.ord.to_u8)
+
+      str.gets_to_end.should eq("bc")
+    end
+
+    it "works with IO#read (already buffered)" do
+      str = IO::Memory.new "abc"
+
+      io = BufferedWrapper.new(str)
+      io.sync?.should be_false
 
-    io.sync = true
-    io.sync?.should be_true
+      byte = Bytes.new(1)
+      io.read_fully(byte)
+      byte[0].should eq('a'.ord.to_u8)
 
-    io.write_byte 1_u8
+      str.gets_to_end.should eq("")
 
-    str.rewind
-    str.read_byte.should eq(1_u8)
+      io.sync = true
+      io.sync?.should be_true
+
+      byte = Bytes.new(1)
+      io.read_fully(byte)
+      byte[0].should eq('b'.ord.to_u8)
```

Comment:
Maybe check that once the buffer is exhausted it will properly resume reading from the IO?

There are 2 independent tests (empty buffer, previously buffered), but the switch isn't tested.

Reply:
Fixed.

Reply:
Why not add some more bytes to the IO::Memory after the first buffered read?

Reply:
Oh, yeah thats a good idea.